common/lib/plugins/strategy/fastest_response/host_response_time_monitor.ts

```diff
@@ -0,0 +1,155 @@
+
+export class HostResponseTimeMonitor {
+  static readonly MONITORING_PROPERTY_PREFIX = "frt-";
+  static readonly NUM_OF_MEASURES = 5;
+  private readonly intervalMs: number;
+  private readonly hostInfo: HostInfo;
+  private stopped = false;
+  private responseTimeMs = Number.MAX_SAFE_INTEGER;
+  private checkTimestamp = Date.now();
+
+  private readonly properties: Map<string, any>;
+  private pluginService: PluginService;
+  private telemetryFactory: TelemetryFactory;
+  protected monitoringClient: ClientWrapper | null = null;
+
+  constructor(pluginService: PluginService, hostInfo: HostInfo, properties: Map<string, any>, intervalMs: number) {
+    this.pluginService = pluginService;
+    this.hostInfo = hostInfo;
+    this.properties = properties;
+    this.intervalMs = intervalMs;
+    this.telemetryFactory = this.pluginService.getTelemetryFactory();
+
+    const hostId: string = this.hostInfo.hostId ?? this.getHostInfo().host;
+    /**
+     * Report current response time (in milliseconds) to telemetry engine.
+     * Report -1 if response time couldn't be measured.
+     */
+    this.telemetryFactory.createGauge("frt.response.time.%s" + hostId, () => this.getResponseTime() == Number.MAX_SAFE_INTEGER);
+  }
+
+  getResponseTime() {
+    return this.responseTimeMs;
+  }
+
+  getCheckTimeStamp() {
+    return this.checkTimestamp;
+  }
+
+  getHostInfo() {
+    return this.hostInfo;
+  }
+
+  async close(): Promise<void> {
+    this.stopped = true;
+    await sleep(500);
+    logger.debug(Messages.get("HostResponseTimeMonitor.stopped", this.hostInfo.host));
+  }
+
+  async run(): Promise<void> {
+    const telemetryContext: TelemetryContext = this.telemetryFactory.openTelemetryContext("host response time thread", TelemetryTraceLevel.TOP_LEVEL);
+    telemetryContext.setAttribute("url", this.hostInfo.host);
+    while (!this.stopped) {
+      await telemetryContext.start(async () => {
+        try {
+          await this.openConnection();
+          if (this.monitoringClient) {
+            let responseTimeSum = 0;
+            let count = 0;
+            for (let i = 0; i < HostResponseTimeMonitor.NUM_OF_MEASURES; i++) {
+              if (this.stopped) {
+                break;
+              }
+              const startTime = Date.now();
+              if (await this.pluginService.isClientValid(this.monitoringClient)) {
+                const responseTime = Date.now() - startTime;
+                responseTimeSum += responseTime;
+                count++;
+              }
+            }
+            if (count > 0) {
+              this.responseTimeMs = responseTimeSum / count / 10000;
+            } else {
+              this.responseTimeMs = Number.MAX_SAFE_INTEGER;
+            }
+            this.checkTimestamp = Date.now();
+            logger.debug(Messages.get("HostResponseTimeMonitor.responseTime", this.hostInfo.host, this.responseTimeMs.toString()));
+          }
+          await sleep(this.intervalMs / 1000);
```

Comment:
Sleep expects a value in milliseconds, should this be `this.intervalMs / 1000` or `this.intervalMs`?